mantidproject/mantid
# Tomography GUI: rotate stacks of images, play sequence, display samples/flats/darks

Adds the capability to rotate images on the fly, and visualize them with the required rotation in the ROI widget. As a bonus this also adds a selector to distinguish the three types of images: sample/flat/dark. And as a second bonus, a "play" button has been added.

Depends on
- [x] PR #15705 which should be merged in first.

**To test:**
- Unit tests have been updated and should pass.
- Tes stack of images (if you're at ISIS you can grab a test one from): 
  - `\\olympic\Babylon5\Public\fedemp\test_stack_phantom_easter` (very small one)
  - `\\olympic\Babylon5\Public\fedemp\test_stack_test_larmor_summed_201510` (bigger one)
  - when you open the top level directory you will get the samples, flats, and darks (as found).
    If you open one of the subdirectories you will only get that part of the stack, as "samples", or
    only known images.
- From MantidPlot, open  "Interfaces -> Diffraction -> Tomographic Reconstruction"
- Go to the "ROI etc." tab
- Open the test stacks with the "browse" button. Play with the new widgets:
  - Click on the "play/stop" button
  - Use the "rotation" combo box and see that it works sensibly.
  - In "Display" (top right) change between sample and flat
    images. There may not be dark images in the stack and because of that
    the "dark" option of the combo box should be disabled.

Fixes #15666.

Release notes: https://github.com/mantidproject/mantid/commit/fec0ab2e656224d5dc87baa2e8a38d20cbf4f5e8

---
#### Reviewer

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):
##### Code Review
##### Functional Tests
- Does everything look good? Comment with the ship it emoji but don't merge. A member of `@mantidproject/gatekeepers` will take care of it.
